src/api/schema/events/metric.rs

```diff
@@ -0,0 +1,78 @@
+use async_graphql::Object;
+use chrono::{DateTime, Utc};
+use vector_core::event::metric::MetricTags;
+
+use super::EventEncodingType;
+use crate::{
+    config::OutputId,
+    event::{self},
+};
+
+#[derive(Debug, Clone)]
+pub struct Metric {
+    output_id: OutputId,
+    event: event::Metric,
+}
+
+impl Metric {
+    pub const fn new(output_id: OutputId, event: event::Metric) -> Self {
+        Self { output_id, event }
+    }
+}
+
+#[Object]
+/// Metric event with fields for querying metric data
+impl Metric {
+    /// Id of the component associated with the metric event
+    async fn component_id(&self) -> &str {
+        self.output_id.component.id()
+    }
+
+    /// Metric timestamp
+    async fn timestamp(&self) -> Option<&DateTime<Utc>> {
+        self.event.data().timestamp().as_ref()
+    }
+
+    /// Metric name
+    async fn name(&self) -> &str {
+        self.event.name()
+    }
+
+    /// Metric namespace
+    async fn namespace(&self) -> Option<&str> {
+        self.event.namespace()
+    }
+
+    /// Metric kind
+    async fn kind(&self) -> &str {
+        match self.event.kind() {
+            event::MetricKind::Incremental => "incremental",
+            event::MetricKind::Absolute => "absolute",
+        }
+    }
+
+    /// Metric type
+    async fn value_type(&self) -> &str {
+        self.event.value().as_name()
+    }
+
+    /// Metric value in human readable form
+    async fn value(&self) -> String {
+        self.event.value().to_string()
+    }
+
+    /// Metric tags
+    async fn tags(&self) -> Option<&MetricTags> {
+        self.event.tags()
+    }
```

Comment:
Actually, I see now tags are rendered as an arbitrary set of key/value pairs in GraphQL. I think the more GraphQL-y approach would be to make `tags` a `[MetricTag!]` where `MetricTag` is

```graphql
type MetricTag {
  key: String!
  value: String!
}
```

Reply:
Updated in [ea4c581](https://github.com/vectordotdev/vector/pull/11201/commits/ea4c5818e1c8ff47780c109a30d3280a8af6da6a)